### ISSUE TITLE
Suggestion: Add a search bar/page?

### ISSUE THREAD

**Issue by thecnoNSMB** [rubyreferences/rubyref#16]
I'm not sure how feasible this would be, given the nature of how the pages are generated, but it would help with usability if I didn't have to manually check every page to see what a `%W()` is, for example.

**Comment by zverok** [comment]
Yes, that's a long-anticipated thing, but how to implement is a bit of question.

Technically, there _are_ solutions for the static sites (they just render the search index into JSON file and load it onto search page, and process search with JS on a client-side), so it is not a big of a problem.

But conceptually, WHAT should be put in a search index, IS a problem. Because if it is "every word in a text", then probably `google term site:rubyreferences.github.io/rubyref` is good enough. So, the task is not to "index it to be searchable", but to create a proper index of terms, which is either a tedious manual work or some inventive automation I haven't decided on yet :)

**Comment by thecnoNSMB** [comment]
Sorry about the inactivity, but you do have a very good point. I'll use the Google method for now and we'll see if someone has a good idea for this eventually.

**Comment by fiveNinePlusR** [comment]
> But conceptually, WHAT should be put in a search index, IS a problem. Because if it is "every word in a text", then probably google term site:rubyreferences.github.io/rubyref

maybe add a search bar that does that on form submission?
